lib/js/jslib.js

```diff
@@ -761,65 +759,60 @@ const LINKS = new (function() {
     for (var i in handlers) {
       var h = handlers[i];
       var hs = eval(h.eventHandlers);
-      resolveServerValue(state, hs);
+      hs = resolveServerValue(state, hs);
       _registerMobileEventHandlers(h.clientKey, hs);
     }
 
     // resolve and spawn the mobile processes
     for (var i in processes) {
-      var p = processes[i];
-      resolveServerValue(state, p);
+      var p = resolveServerValue(state, processes[i]);
       _spawnWithMessages(p.pid, p.process, p.messages);
     }
     return;
   }
 
 
   /**
-   * Resolve function references in the object `obj`, specified as records
-   * { function:f, environment:e }, where the environment is optional. If
-   * an environment is specified, we assume that the function denoted by
+   * Resolve function references in the object `obj` and return the
```

Comment:
You should note here that the resolution process fails for deeply nested structures.

Reply:
BTW you can fix this problem in a follow up PR.

Reply:
I have added the comment.